### PR TITLE
[ffmpeg] Some libraries in ffmpeg can be optional

### PR DESCRIPTION
**Pull request by sandercox**
I often use FFmpeg in my code but find no use for some sublibraries. I've created config options to optionally enable these libraries when required.

avcodec
avdevice
avfilter
avformat
swresample
swscale

### CODE REVIEW

File: ports/ffmpeg/CONTROL [hunk 1]
```diff
@@ -5,7 +5,7 @@ Build-Depends: zlib
 Homepage: https://ffmpeg.org
 Description: a library to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created.
   FFmpeg is the leading multimedia framework, able to decode, encode, transcode, mux, demux, stream, filter and play pretty much anything that humans and machines have created. It supports the most obscure ancient formats up to the cutting edge. No matter if they were designed by some standards committee, the community or a corporation. It is also highly portable: FFmpeg compiles, runs, and passes our testing infrastructure FATE across Linux, Mac OS X, Microsoft Windows, the BSDs, Solaris, etc. under a wide variety of build environments, machine architectures, and configurations.
-Default-Features: avresample
+Default-Features: avresample, avdevice, avfilter, swresample, swscale
```
Comment (NancyLi1013) [inline ports/ffmpeg/CONTROL:8]: Why do you set these new features as default features?

Comment (NancyLi1013) [reply ports/ffmpeg/CONTROL:8]: Please update `Port-Version ` from 17 to 18.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by mcmtroffaes** [comment]
This is very useful to help reducing compile times on ffmpeg. Support for disabling avcodec could also be added, for the sake of completeness (but this patch is fine without it, IMO).

@NancyLi1013 I believe these new options should be enabled by default, as they are built by default by upstream (and also were always built prior to this patch, since they were implicitly enabled).

**Comment by sandercox** [comment]
@mcmtroffaes for sake of completeness I also added avcodec and avformat.
Thanks for pointing that out.

@mcmtroffaes @NancyLi1013 indeed these now optional features used to be default so that was my reason for putting them on the default list.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by mcmtroffaes** [comment]
Thanks for adding avcodec & avformat as well @sandercox - that's great!

File: ports/ffmpeg/FindFFMPEG.cmake.in [hunk 2]
```diff
@@ -131,32 +131,68 @@ if(APPLE)
...
+    list(APPEND FFMPEG_INCLUDE_DIRS ${FFMPEG_libswscale_INCLUDE_DIRS})
+  endif()
   list(REMOVE_DUPLICATES FFMPEG_INCLUDE_DIRS)
   list(REMOVE_DUPLICATES FFMPEG_LIBRARY_DIRS)
 
-  set(FFMPEG_libavcodec_VERSION "${FFMPEG_VERSION}" CACHE STRING "")
-  set(FFMPEG_libavdevice_VERSION "${FFMPEG_VERSION}" CACHE STRING "")
-  set(FFMPEG_libavfilter_VERSION "${FFMPEG_VERSION}" CACHE STRING "")
-  set(FFMPEG_libavformat_VERSION "${FFMPEG_VERSION}" CACHE STRING "")
+  if(FFMPEG_libavdevice_FOUND)
```
Comment (mcmtroffaes) [inline ports/ffmpeg/FindFFMPEG.cmake.in:174]: I think this needs to be ``if(FFMPEG_libavcodec_FOUND)``?

Comment (NancyLi1013) [inline ports/ffmpeg/FindFFMPEG.cmake.in:174]:
```suggestion
  if(FFMPEG_libavcodec_FOUND)
```

Comment (NancyLi1013) [reply ports/ffmpeg/FindFFMPEG.cmake.in:174]: Thanks for your reminder. Sorry,I didn't notice the point before this.

Comment (mcmtroffaes) [reply ports/ffmpeg/FindFFMPEG.cmake.in:174]: No problem. I'm sorry too for not noticing when I first looked over it.

**Comment by sandercox** [comment]
Thanks for pointing out my obvious copy & paste bug. Should all be fixed now.

**Pull request merged**